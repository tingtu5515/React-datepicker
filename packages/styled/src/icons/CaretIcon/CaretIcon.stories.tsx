import React from 'react'
import {storiesOf} from '@storybook/react'
import {text, color, select} from '@storybook/addon-knobs'
import CaretIcon from '.'

const IconDirections = {
  left: 'left',
  right: 'right',
  up: 'up',
  down: 'down',
}

storiesOf('Components/Icons', module).add('CaretIcon', () => (
  <CaretIcon
    color={color('Color', '#000')}
    height={text('Height', '30px')}
    width={text('Width', '30px')}
    // @ts-ignore
    direction={select('Direction', IconDirections, 'right')}
  />
))
