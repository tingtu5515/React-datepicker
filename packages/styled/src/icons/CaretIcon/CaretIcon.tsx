import React from 'react'

interface Props extends IconProps {
  direction?: 'up' | 'down' | 'left' | 'right'
}

function calculateAngle(direction: Props['direction']) {
  switch (direction) {
    case 'up':
      return 180
    case 'down':
      return 0
    case 'left':
      return 90
    case 'right':
    default:
      return -90
  }
}

function CaretIcon({height, width, color, direction = 'right', className = ''}: Props) {
  const angle = calculateAngle(direction)
  return (
    <svg
      width={width}
      height={height}
      className={className}
      color={color}
      transform={`rotate(${angle} 0 0)`}
      viewBox="0 0 9 6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill="currentColor"
        fillRule="evenodd"
        d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      />
    </svg>
  )
}

export default CaretIcon
