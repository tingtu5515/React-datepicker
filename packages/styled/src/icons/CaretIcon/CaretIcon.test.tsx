import * as React from 'react'
import {render} from '@testing-library/react'
import CaretIcon from '.'

test('should match snapshot', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" />)
  expect(container).toMatchSnapshot()
})

test('should get classname', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" className="test" />)
  expect(container).toMatchSnapshot()
})

test('should be facing to the left', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" direction="left" />)
  expect(container).toMatchSnapshot()
})

test('should be facing to the right', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" direction="right" />)
  expect(container).toMatchSnapshot()
})

test('should be facing down', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" direction="down" />)
  expect(container).toMatchSnapshot()
})

test('should be facing up', () => {
  const {container} = render(<CaretIcon height="30px" width="30px" color="red" direction="up" />)
  expect(container).toMatchSnapshot()
})
