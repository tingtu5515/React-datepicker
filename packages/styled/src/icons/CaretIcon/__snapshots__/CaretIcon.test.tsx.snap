// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be facing down 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="30px"
    transform="rotate(0 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`should be facing to the left 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="30px"
    transform="rotate(90 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`should be facing to the right 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="30px"
    transform="rotate(-90 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`should be facing up 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="30px"
    transform="rotate(180 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`should get classname 1`] = `
<div>
  <svg
    class="test"
    color="red"
    height="30px"
    transform="rotate(-90 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`should match snapshot 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="30px"
    transform="rotate(-90 0 0)"
    viewBox="0 0 9 6"
    width="30px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.058 4.594L1.185 1.72a.312.312 0 1 1 .442-.442L4.5 4.152l2.873-2.873a.312.312 0 1 1 .442.442L4.723 4.812a.316.316 0 0 1-.446 0l-.219-.218z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
